IBX-2829: Added option to configure fields for register form (#1663)
ezsystems/developer-documentation: docs/guide/user_management/user_management.md

You can allow your users to create accounts by employing the `/register` route. This route leads to a registration form that, when filled in, creates a new User Content item in the repository.

### User types

> To decide to what workspace the user should be registered to, you need to specify their user type under `user_type_identifier`.

Reviewer: We don't use "workspace" anywhere else, either the term needs explanation, or rephrase.

> Currently, there are two user types defined: `users` and `customers`.

Reviewer: ```suggestion
There are two user types defined: `users` and `customers`.
```

> `users` are backend users that are involved in creating the page such as editors and `customers` are frontend users.

Reviewer: ```suggestion
`users` are Back Office users that are involved in creating the page such as editors, and `customers` are frontend users.
```

```yaml
ibexa:
    system:

>         <siteaccess>:

Reviewer: ```suggestion
        <scope>:
```

                group_id: <userGroupContentId>
```

### Registration form field configuration

> To modify the registration form template add or remove fields under `allowed_field_definitions_identifiers`:

Reviewer: ```suggestion
To modify the registration form template, add or remove fields under `allowed_field_definitions_identifiers`:
```

```yaml
ibexa:
    system:

>         <siteaccess>:

Reviewer: ```suggestion
        <scope>:
```

            user_registration:
                user_type_identifier: user
                form:
                    allowed_field_definitions_identifiers:
                        - first_name
                        - last_name
                        - user_account
```

### Other user management templates

> You can also modify form templates in this way:

Reviewer: ```suggestion
You can also modify form templates in the following way:
```
